aptos-move/block-executor/src/scheduler.rs

```diff
@@ -362,7 +363,7 @@ impl Scheduler {
                             // All txns have been committed, the parallel execution can finish.
                             self.done_marker.store(true, Ordering::SeqCst);
                         }
-                        return Some(*commit_idx - 1);
+                        return Some((*commit_idx - 1, incarnation + 1));
```

Comment:
I would return incarnation (that's the committed incarnation here I think for try_commit), and then increase it in the caller for re-execution?